sdk/scheduler/src/main/java/com/mesosphere/sdk/scheduler/SchedulerConfig.java

```diff
@@ -351,6 +356,13 @@ public boolean isregionAwarenessEnabled() {
         return Boolean.valueOf(envStore.getOptional(ALLOW_REGION_AWARENESS_ENV, "false"));
     }
 
+    /**
+     * Returns an optional of the custom Service TLD.
+     */
+    public Optional<String> getCustomServiceTLD() {
```

Comment:
As already mentioned, would a `getTLD` method that returns a `String` make sense instead?

Reply:
I don't want to split half of the functionality of EndpointUtils into SchedulerConfig.

And so here it needs to be an Optional as the _absence_ of the custom TLD is meaningful.

Reply:
OK. It was just not clear that the absence was meaningful, as the only time the absence is used to make a decision is to revert to the default.